### PR TITLE
Added SIMD-accelerated restricted Damerau-Levenshtein distance for BK-tree

### PR DESCRIPTION
**Pull request by Daniel-Liu-c0deb0t**
I have modified your code to use a SIMD-accelerated version of the OSA distance. It passes all tests, but OSA is not a metric so the BK-tree may produce wrong results. It is better to use Levenshtein distance instead. I can edit the code to do that if you want.

### CODE REVIEW

**Comment by oddcoder** [comment]
@Daniel-Liu-c0deb0t I just realized that it it might break the tree logic, true, Feel free to use  Levenshtein instead.

File: rtrees/src/bktree/tree.rs [hunk 1]
```diff
@@ -111,6 +117,7 @@ where
     }
 }
 
+#[allow(unused)]
```
Comment (oddcoder) [inline rtrees/src/bktree/tree.rs]: Feel free to completely remove this function, now that it is completely replaced.

**Comment by Daniel-Liu-c0deb0t** [comment]
I've removed the OSA code and swapped it to Levenshtein distance. I had to update the test cases.

Let me know if you want other distance metrics. Given the current use case, Levenshtein distance is probably fine.

**Comment by oddcoder** [comment]
thank you, Merged!

**Pull request merged**